### PR TITLE
Add simple overloads for QueryKnowledgeBase

### LINKED ISSUE #24172: Make deploymentName in QuestionAnsweringClient required
The Question Answering team has decided they will make `deploymentName` required like CLU has already, but until #24067 is complete and the swagger updated, go ahead and refactor the APIs to assume `deploymentName` already is required, much like in PR #24030

/cc @rokulka @annatisch @iscai-msft

### PR DESCRIPTION
**Pull request by heaths**
Resolves #24172. Also regenerated with transform swagger as needed for #24067, but that is waiting on the final expected swagger changes.

### CODE REVIEW

**Comment by heaths** [comment]
16 of the files are just re-recordings, though I recommend at least sampling changes in a few. I aligned the env vars to almost what @annatisch had, though the "AZURE_" prefix is unnecessary for .NET because we fall back to checking that as well.

File: sdk/cognitivelanguage/Azure.AI.Language.QuestionAnswering/tests/Samples/Readme.cs [hunk 1]
```diff
@@ -24,12 +24,11 @@ public void CreateQuestionAnsweringClient()
         public void BadArgument()
         {
             QuestionAnsweringClient client = Client;
-            QueryKnowledgeBaseOptions options = new QueryKnowledgeBaseOptions("Does this knowledge base exist?");
 
             #region Snippet:QuestionAnsweringClient_BadRequest
             try
             {
-                Response<KnowledgeBaseAnswers> response = client.QueryKnowledgeBase("invalid-knowledgebase", options);
+                Response<KnowledgeBaseAnswers> response = client.QueryKnowledgeBase("invalid-knowledgebse", "test", "Does this knowledge base exist?");
```
Comment (AhmedLeithy) [inline sdk/cognitivelanguage/Azure.AI.Language.QuestionAnswering/tests/Samples/Readme.cs:31]: Nit: knowledgebase has an 'a' missing.

File: sdk/cognitivelanguage/Azure.AI.Language.QuestionAnswering/README.md [hunk 2]
```diff
@@ -126,7 +129,7 @@ For example, if you submit a question to a non-existant knowledge base, a `400`
 ```C# Snippet:QuestionAnsweringClient_BadRequest
 try
 {
-    Response<KnowledgeBaseAnswers> response = client.QueryKnowledgeBase("invalid-knowledgebase", options);
+    Response<KnowledgeBaseAnswers> response = client.QueryKnowledgeBase("invalid-knowledgebse", "test", "Does this knowledge base exist?");
```
Comment (JoshLove-msft) [inline sdk/cognitivelanguage/Azure.AI.Language.QuestionAnswering/README.md:132]:
```suggestion
    Response<KnowledgeBaseAnswers> response = client.QueryKnowledgeBase("invalid-knowledgebase", "test", "Does this knowledge base exist?");
```

**Review by JoshLove-msft — APPROVED** [review]

**Pull request merged**